nodes/ui_base.html

```diff
@@ -1874,11 +1874,12 @@
                         }
                     }
                     // buttons
+                    var nodeId = item.node.id.split('.');
                     var buttonGroup = $('<div>',{class:"nr-db-sb-list-header-button-group"}).appendTo(titleRow);
                     if (item.type === 'ui_tab') {
-                        var addGroupButton = $('<a href="#" class="editor-button editor-button-small nr-db-sb-list-header-button"><i class="fa fa-plus"></i> '+c_("layout.group")+'</a>').appendTo(buttonGroup);
+                        var addGroupButton = $('<a href="#" class="tab-add-group-button-' + nodeId[0] + '-' + nodeId[1] +' editor-button editor-button-small nr-db-sb-list-header-button" ><i class="fa fa-plus"></i> '+c_("layout.group")+'</a>').appendTo(buttonGroup);
```

Comment:
Why are you adding the id into the class ? I thought your were using `node-id= ` for the nodeid ? Or indeed as node.id is unique it could be `id= `  
If it has to be class then can we make them consistent with other dashboard sidebar classes ie start `nr-db-sb-`

Reply:
Hi

I will explain:

1. Work of Editor heavily based on DOM emulation, like: call up the right panel, edit for node, group, tab. So adding a class with nodeid is covenient way to call. Split function of nodeId for class because dot (.) is not supported in class name.
2. node-id attribute is added as default for editor used
3. nr-db-sb-list-header-button: This class used for both add space and edit group button. I am using `editor-button-edit-group` to quickly calculate the number of current group. If you tend to remove this, I will have to rewrite the calculate function.

Reply:
1) OK (about dot) -but why not use id as the mode.id are unique ? getElementById would  target it exactly.
2) We only added node-id to elements in the dashboard. Don't forget - This is the editor - they are not the same web page - so we don't need them here then (If using class or id).
3) All editor- classes are part of the core Node-RED editor. The Dashboard plugin should not be creating classes (or ids) that may overlap so should be easily identified as being part of the dashboard - hence use names starting with nr-db-sb-...

Reply:
Tks for clear out. I will make fix based on this discussion in ui base:
1. Change node-id in ui base to id
2. Modify necessary class with prefix nr-db-sb

Will submit a PR again. Tks

Reply:
Well if you use id - then do you need class as well ? the id will target it precisely. Class should really be used if you want to target more than one.

Reply:
- Button group for `Tab` has 3 button (add group, edit tab, edit layout), and I call them all so need class with ID is required to hook up. I put id in edit button.
- Button group for `Group` has 2 button (add spacer, edit), so adding class for future use with spacer.

Reply:
you just need to edit and push the original PR - it should automatically sync up.
But I'm afraid I still don't get the logic.  Either use id=.... and the node.id to target an individual button. OR use the class across all buttons of the group as that then targets all the ones in the group. If you need classes per button then again they can be identical eg nr-db-sb-tab-layout and the id at the groupButton level above to tell them apart - that is how css is supposed to work isn't it ?

Reply:
I got your point. I made a sample html so could you review before I made real modification. Because each time changing in ui_base, I must make many changes in Editor, so hope you could help to clear out:

Tab group button
```html
<div id="tabId" class="nr-db-sb-list-header-button-group">
    <a href="#" class="nr-db-sb-tab-add-group-button editor-button editor-button-small nr-db-sb-list-header-button">
        <i class="fa fa-plus"></i> group</a>
    <a href="#" id="afd9c0f1.b47828"
        class="nr-db-sb-tab-edit-button editor-button  editor-button-small nr-db-sb-list-header-button">
        <i class="fa fa-pencil"></i> edit</a>
    <a href="#" class="nr-db-sb-tab-edit-layout-button editor-button editor-button-small nr-db-sb-list-header-button">
        <i class="fa fa-pencil"></i> layout</a>
</div>
````
Group button
```html
<div id="groupId" class="nr-db-sb-list-header-button-group">
    <a href="#" class="editor-button editor-button-small nr-db-sb-list-header-button"><i class="fa fa-plus"></i>
        spacer</a>
    <a href="#" 
        class="nr-db-sb-edit-group-button editor-button editor-button-small nr-db-sb-list-header-button">
        <i class="fa fa-pencil"></i> edit</a>
</div>
```

Reply:
Hi

I make a new commit with template as previous post. ID is in group, button has sepearate class name